### DISCUSSION TITLE
How to query RHI (inside CrashContext.runtime-xml) from Sentry

### DISCUSSION THREAD

**Discussion by joshuaclausen** [getsentry/sentry-unreal#234] [General]
I would like to query RHI fields in the CrashContext.runtime-xml file that the Unreal Crash Reporter sends to Sentry.  I can see the file, and I can view the file as expected.  However, it would amazing if I could somehow query the data via a Sentry query.  Imagine being able to show a chart of crashes by RHI.UserDriverVersion.

It seems I would have to tell Sentry to index some of the fields in that XML file.  It seems that might have to be done at crash ingestion time, rather than downloading and scanning every CrashContext.runtime-xml file at query time.

Where might I start with this?

**Comment by tustanivsky** [comment]
Maybe @bitsandfoxes can point you in the right direction?

**Comment by joshuaclausen** [comment]
Just some further thoughts on the topic.

I can see there is the option to assign custom tags to the Crash Reporter: https://docs.sentry.io/platforms/unreal/setup-crashreporter/#configure-attributes.

In order to leverage CrashContext.runtime-xml data that the Unreal Crash Reporter is already sending to Sentry, it seems Sentry would have to have an ability to scan that incoming file that the Unreal Engine Crash Reporter is already sending and generate tags.  It's not clear to me if that's the case, or if I'm really looking at having to set custom tags in the client code.  Any recommendations on that topic would be appreciated.

I'm storing all crash reports and files in an S3 bucket, so I suppose another option might be to somehow use Clickhouse to query those files and generate reports.  However, I suspect this would be entirely outside of Sentry.
